package NAntiRobot.NFeaturesProto;

message THeader {
    optional string Header = 1;
    optional uint64 Num = 2;
};

message TFloatRecord {
    optional string Key = 1;
    optional float Value = 2;
};

message TStringRecord {
    optional string Key = 1;
    optional string Value = 2;
};

message TFixed64Record {
    optional string Key = 1;
    optional fixed64 Value = 2;
};

message TCityHash64FloatRecord {
    optional fixed64 Key = 1;
    optional float Value = 2;
};

message TUidRecord {
    optional fixed64 Key = 1;
};

message TMarketJwsStatesStats {
    optional string Key = 1;
    optional float DefaultExpiredRatio = 2;
    optional float DefaultRatio = 3;
    optional float JwsStateIsInvalidRatio = 4;
    optional float SuspExpiredRatio = 5;
    optional float SuspRatio = 6;
    optional float ValidExpiredRatio = 7;
    optional float ValidRatio = 8;
};

message TMarketStats {
    optional fixed64 Key = 1;
    optional uint64 BlockedCntRatio = 2;
    optional uint64 CatalogReqsCntRatio = 3;
    optional uint64 EnemyCntRatio = 4;
    optional uint64 EnemyRedirectsCntRatio = 5;
    optional uint64 FuidCntRatio = 6;
    optional uint64 HostingCntRatio = 7;
    optional uint64 IcookieCntRatio = 8;
    optional uint64 Ipv4CntRatio = 9;
    optional uint64 Ipv6CntRatio = 10;
    optional uint64 LoginCntRatio = 11;
    optional uint64 MobileCntRatio = 12;
    optional uint64 OtherHandlesReqsCntRatio = 13;
    optional uint64 ProductReqsCntRatio = 14;
    optional uint64 ProxyCntRatio = 15;
    optional uint64 RefererIsEmptyCntRatio = 16;
    optional uint64 RefererIsNotYandexCntRatio = 17;
    optional uint64 RefererIsYandexCntRatio = 18;
    optional uint64 RobotsCntRatio = 19;
    optional uint64 SearchReqsCntRatio = 20;
    optional uint64 SpravkaCntRatio = 21;
    optional uint64 TorCntRatio = 22;
    optional uint64 VpnCntRatio = 23;
    optional uint64 YndxIpCntRatio = 24;
};
